test_pull_requests

```diff
@@ -420,27 +424,30 @@ class InvalidatingGitHubCache < Faraday::HttpCache
 
     attr_accessor :invalidatable
 
+    require 'set'
+
     def initialize(*args)
       super
-      @invalidatable = Hash.new { |h,k| h[k] = {} }
+      @invalidatable = Hash.new { |h,k| h[k] = Set.new }
       @cache.storage = self
     end
 
-    # GitHub-specific (although probably other APIs are similar) method
-    # to remove the pagination parameters from a URL so it can be better
-    # classified
+    # GitHub-specific (although probably other APIs are similar)
+    # method to remove specific resource IDs from URLs so they can be
+    # matched to the cooresponding resource fetch URLS
     #
     # Example:
-    #   synthesize_url("https://api.github.com/some/resource/id?per_page=100")
+    #   derive_api_match_url("https://api.github.com/some/resource/id?per_page=100")
     #     returns
     #   "https://api.github.com/some/resource/id"
-    def synthesize_url(orig_url)
+    def derive_api_match_url(orig_url=nil)
+      return nil if orig_url.nil?
```

Comment:
Yeah, I put that in to save some logic in the caller, thought better of it and forgot to remove it.